qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -1326,6 +1331,8 @@ def _set_service(self, service: DatabaseServiceV1) -> None:
         if self._service:
             raise DbExperimentDataError("An experiment service is already being used.")
         self._service = service
+        for result in self._analysis_results.values():
+            result.service = service
```

Comment:
If the service of the results is not getting set correctly when loading from DB rather than fix it here it might be better to set the service in the `DbExperimentData._retrieve_analysis_results` or `DbAnalysisResult._from_service_data` since these will also affect other functions that retrieve results from DB, not just load.

Reply:
I don't understand. Setting the service here, when the experiment service is set, will affect any function that will query about the service, not just `load`.